app/dts/behaviors/capslock.dtsi

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <dt-bindings/zmk/keys.h>
+
+/ {
+	behaviors {
+		/omit-if-no-ref/ capslock_on: behavior_capslock_on {
+			compatible = "zmk,behavior-capslock";
+			label = "CAPSLOCK_ON";
+			#binding-cells = <0>;
+			capslock-press-duration = <5>;
+			enable-on-press;
+		};
+		
+		/omit-if-no-ref/ capslock_off: behavior_capslock_off {
+			compatible = "zmk,behavior-capslock";
+			label = "CAPSLOCK_OFF";
+			#binding-cells = <0>;
+			capslock-press-duration = <5>;
+			disable-on-release;
+		};
+		
+		/omit-if-no-ref/ capslock_hold: behavior_capslock_hold {
+			compatible = "zmk,behavior-capslock";
+			label = "CAPSLOCK_HOLD";
+			#binding-cells = <0>;
+			capslock-press-duration = <5>;
+			enable-on-press;
+			disable-on-release;
+		};
+		
+		/omit-if-no-ref/ capslock_word: behavior_capslock_word {
+			compatible = "zmk,behavior-capslock";
+			label = "CAPSLOCK_WORD";
+			#binding-cells = <0>;
+			capslock-press-duration = <5>;
+			enable-on-press;
+			disable-on-next-release;
+			disable-on-keys = <SPACE TAB ENTER ESC>;
```

Comment:
This is inconsistent with `&caps_word`, which uses a list of keys that _continue_ a word rather than end it. There are many other keys (such as punctuation) that I would probably expect to end a word.

As I noted on the PR as a whole, I think it would be confusing to have this and `&caps_word` behave differently. It would likely be easier to keep the two in sync if they were literally the same behavior, but with a property to switch to using caps lock instead of modifiers.